interface/ceed-elemrestriction.c

```diff
@@ -574,6 +588,33 @@ int CeedElemRestrictionCreateBlockedStrided(Ceed ceed, CeedInt num_elem,
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Copy the pointer to a CeedElemRestriction. Both pointers should
+           be destroyed with `CeedElemRestrictionDestroy()`;
+           Note: If `*rstr_copy` is non-NULL, then it is assumed that
+           `*rstr_copy` is a pointer to a CeedElemRestriction. This
+           CeedElemRestriction will be destroyed if `*rstr_copy` is the
+           only reference to this CeedElemRestriction.
+
+  @param rstr            CeedElemRestriction to copy reference to
+  @param[out] rstr_copy  Variable to store copied reference
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref User
+**/
+int CeedElemRestrictionCopyReference(CeedElemRestriction rstr,
+                                     CeedElemRestriction *rstr_copy) {
+  int ierr;
+
+  ierr = CeedElemRestrictionIncrementRefCounter(rstr); CeedChk(ierr);
+  if (*rstr_copy) {
+    ierr = CeedElemRestrictionDestroy(rstr_copy); CeedChk(ierr);
+  }
```

Comment:
We shouldn't need the guard because destroy checks for NULL.
```c
  if (!*rstr || --(*rstr)->ref_count > 0) return CEED_ERROR_SUCCESS;
```

Reply:
Duh, I should have seen that. Thank you